packages/core/src/transports/base.ts

```diff
@@ -40,14 +42,18 @@ export function createTransport(
   ),
 ): Transport {
   let rateLimits: RateLimits = {};
-
   const flush = (timeout?: number): PromiseLike<boolean> => buffer.drain(timeout);
+  const client = getCurrentHub().getClient();
 
   function send(envelope: Envelope): PromiseLike<void | TransportMakeRequestResponse> {
     const filteredEnvelopeItems: EnvelopeItem[] = [];
 
     // Drop rate limited items from envelope
     forEachEnvelopeItem(envelope, (item, type) => {
+      if (client && client.emit) {
+        client.emit('beforeEnvelope', envelope);
```

Comment:
shouldn't this be called above `forEachEnvelopeItem`?

Reply:
Ah right, idk why I placed it here but yes, it def should be. Do I even need to feature check here btw, can I just client.emit...?

Reply:
`emit` may be unavailable when using a custom client, so until v8 we need to guard this, sadly!